src/afSubmit.js

```diff
@@ -6,24 +6,20 @@ angular.module('angularFormMessages').directive('afSubmit', function () {
     controller: function afSubmitController($scope) {
       this.validations = {};
 
-      this.validate = function (modelPath, isValid, message) {
-        this.validations[modelPath] = {
-          isValid: isValid,
-          message: message
-        };
+      this.validate = function (modelPath, errors) {
+        this.validations[modelPath] = errors;
         $scope.validations = this.validations; // Temp
-        $scope.$broadcast('validation', modelPath, isValid, message);
+        $scope.$broadcast('validation', modelPath, errors);
       };
 
       this.isValid = function () {
-        var isValid = true;
-        angular.forEach(this.validations, function (validation) {
-          if (!validation.isValid) {
-            isValid = false;
-            return;
+        for (var key in this.validations) {
+          var errors = this.validations[key];
```

Comment:
Waarom eigenlijk op `key` itereren? Zoals het in processErrors gebeurt lijkt me logischer. Of doe je het zo omdat je dan meteen kunt returnen?